trackers/core/ksptracker/tracker.py

```diff
@@ -0,0 +1,277 @@
+                    future_dets = all_detections[future_idx]
+                    for future_det_idx in range(len(future_dets)):
+                        if future_dets.confidence[future_det_idx] < self.min_confidence:
+                            continue
+
+                        future_node = TrackNode(
+                            frame_id=future_idx,
+                            detection_id=future_det_idx,
+                            bbox=tuple(future_dets.xyxy[future_det_idx]),
+                            confidence=future_dets.confidence[future_det_idx],
+                        )
+
+                        if self._can_connect_nodes(node, future_node):
+                            weight = self._edge_cost(node, future_node)
+                            G.add_edge(node, future_node, weight=weight)
+
+        return G
+
+    def _update_detections_with_tracks(self, assignments: Dict) -> sv.Detections:
+        """Update detections with track IDs based on assignments.
+
+        Args:
+            assignments (Dict): Maps (frame_id, det_id) to track_id
+
+        Returns:
+            sv.Detections: Updated detections with tracker_ids assigned
+        """
+        all_detections = []
+        all_tracker_ids = []
+
+        for frame_idx, dets in enumerate(self.detection_buffer):
+            frame_tracker_ids = [-1] * len(dets)
+
+            for det_idx in range(len(dets)):
+                key = (frame_idx, det_idx)
+                if key in assignments:
+                    frame_tracker_ids[det_idx] = assignments[key]
+
+            all_detections.append(dets)
+            all_tracker_ids.extend(frame_tracker_ids)
+
+        final_detections = sv.Detections.merge(all_detections)
+        final_detections.tracker_id = np.array(all_tracker_ids)
+
+        return final_detections
+
+    def ksp(self, graph: nx.DiGraph) -> List[List[TrackNode]]:
+        """Find K-shortest paths in the graph.
+
+        Args:
+            graph (nx.DiGraph): Directed graph of detection nodes
+
+        Returns:
+            List[List[TrackNode]]: List of paths, each path is list of TrackNodes
+        """
+        paths: List[List[TrackNode]] = []
+        for path in nx.shortest_simple_paths(graph, "source", "sink", weight="weight"):
+            if len(paths) >= self.max_paths:
+                break
+            # Remove source and sink nodes from path
+            paths.append(path[1:-1])
+        return paths
+
+    def process_tracks(self) -> sv.Detections:
+        """Process all buffered detections to create final tracks.
+
+        Returns:
+            sv.Detections: Detections with assigned track IDs
+        """
+        graph = self._build_graph(self.detection_buffer)
+        paths = self.ksp(graph)
+
+        # Assign track IDs
+        assignments = {}
+        for track_id, path in enumerate(paths, start=1):
+            for node in path:
+                assignments[(node.frame_id, node.detection_id)] = track_id
```

Comment:
This simply overwrites the `track_id`, meaning only the last path found containing that detection gets the assignment. This doesn't seem like an accurate representation of the disjoint-path formulation described in the paper.